### ISSUE TITLE
InternVL3运行GKD算法报错RuntimeError: GET was unable to find an engine to execute this computation

### ISSUE THREAD

**Issue by QiZishi** [modelscope/ms-swift#6628]
训练脚本如下：
NPROC_PER_NODE=6 \
PYTORCH_CUDA_ALLOC_CONF='expandable_segments:True' \
CUDA_VISIBLE_DEVICES=1,2,3,4,5,6 \
swift rlhf \
    --rlhf_type gkd \
    --model OpenGVLab/InternVL3-2B \
    --teacher_model DAMO_Academy/Lingshu-7B \
    --teacher_model_type qwen2_5_vl \
    --train_type full \
    --dataset xxx \
    --seq_kd false \
    --lmbda 1 \
    --beta 1 \
    --torch_dtype bfloat16 \
    --num_train_epochs 1 \
    --per_device_train_batch_size 1 \
    --learning_rate 1e-5 \
    --gradient_accumulation_steps 1 \
    --save_steps 100 \
    --save_total_limit 2 \
    --logging_steps 1 \
    --max_length 16000 \
    --max_completion_length 8192 \
    --output_dir xxxx \
    --warmup_ratio 0.05 \
    --save_only_model true \
    --dataloader_num_workers 64 \
    --dataset_num_proc 4 \
    --deepspeed zero2 \
    --teacher_deepspeed zero3 \
    --offload_teacher_model true \
    --log_completions true \
    --attn_impl flash_attention_2 \
    --use_vllm true \
    --vllm_limit_mm_per_prompt '{"image":5,"video":0}'\
    --vllm_mode server \
    --vllm_server_host 127.0.0.1 \
    --vllm_server_port 8000 \

报错代码如下：
  File "/home/qzs/conclusion/ms-swift/swift/cli/rlhf.py", line 7, in <module>
    rlhf_main()
  File "/home/qzs/conclusion/ms-swift/swift/llm/train/rlhf.py", line 233, in rlhf_main
    return SwiftRLHF(args).main()
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/qzs/conclusion/ms-swift/swift/llm/base.py", line 49, in main
    result = self.run()
             ^^^^^^^^^^
  File "/home/qzs/conclusion/ms-swift/swift/ray/base.py", line 170, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/qzs/conclusion/ms-swift/swift/llm/train/sft.py", line 196, in run
    return self.train(trainer)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/qzs/conclusion/ms-swift/swift/llm/train/sft.py", line 244, in train
    trainer.train(trainer.args.resume_from_checkpoint)
  File "/home/qzs/conclusion/ms-swift/swift/trainers/mixin.py", line 815, in train
    res = super().train(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/env_swift/lib/python3.12/site-packages/transformers/trainer.py", line 2206, in train
    return inner_training_loop(
           ^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/env_swift/lib/python3.12/site-packages/transformers/trainer.py", line 2548, in _inner_training_loop
    tr_loss_step = self.training_step(model, inputs, num_items_in_batch)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/qzs/conclusion/ms-swift/swift/trainers/rlhf_trainer/utils.py", line 407, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/qzs/conclusion/ms-swift/swift/trainers/rlhf_trainer/gkd_trainer.py", line 241, in training_step
    loss = HFSFTTrainer.training_step(self, model, inputs, num_items_in_batch)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/env_swift/lib/python3.12/site-packages/trl/trainer/sft_trainer.py", line 733, in training_step
    return super().training_step(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/env_swift/lib/python3.12/site-packages/transformers/trainer.py", line 3749, in training_step
    loss = self.compute_loss(model, inputs, num_items_in_batch=num_items_in_batch)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/qzs/conclusion/ms-swift/swift/trainers/rlhf_trainer/utils.py", line 407, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/qzs/conclusion/ms-swift/swift/trainers/rlhf_trainer/gkd_trainer.py", line 130, in compute_loss
    outputs_student = model(**model_inputs)
                      ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/env_swift/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1751, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/env_swift/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1762, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/env_swift/lib/python3.12/site-packages/deepspeed/utils/nvtx.py", line 20, in wrapped_fn
    ret_val = func(*args, **kwargs)
              ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/env_swift/lib/python3.12/site-packages/deepspeed/runtime/engine.py", line 2054, in forward
    loss = self.module(*inputs, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/env_swift/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1751, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/env_swift/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1857, in _call_impl
    return inner()
           ^^^^^^^
  File "/opt/anaconda3/envs/env_swift/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1784, in inner
    args_kwargs_result = hook(self, args, kwargs)  # type: ignore[misc]
                         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/qzs/conclusion/ms-swift/swift/llm/template/base.py", line 1351, in pre_forward_hook
    kwargs = to_device(self._post_encode(model, old_kwargs), model.device)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/qzs/conclusion/ms-swift/swift/llm/template/template/internvl.py", line 82, in _post_encode
    vit_embeds = model.extract_feature(pixel_values).to(device=device)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/qzs/.cache/huggingface/modules/transformers_modules/InternVL3-2B/modeling_internvl_chat.py", line 186, in extract_feature
    vit_embeds = self.vision_model(
                 ^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/env_swift/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1751, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/env_swift/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1762, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/qzs/.cache/huggingface/modules/transformers_modules/InternVL3-2B/modeling_intern_vit.py", line 412, in forward
    hidden_states = self.embeddings(pixel_values)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/env_swift/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1751, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/env_swift/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1857, in _call_impl
    return inner()
           ^^^^^^^
  File "/opt/anaconda3/envs/env_swift/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1805, in inner
    result = forward_call(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/qzs/.cache/huggingface/modules/transformers_modules/InternVL3-2B/modeling_intern_vit.py", line 164, in forward
    patch_embeds = self.patch_embedding(pixel_values)  # shape = [*, channel, width, height]
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/env_swift/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1751, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/env_swift/lib/python3.12/site-packages/torch/nn/modules/module.py", line 1762, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/env_swift/lib/python3.12/site-packages/torch/nn/modules/conv.py", line 554, in forward
    return self._conv_forward(input, self.weight, self.bias)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/anaconda3/envs/env_swift/lib/python3.12/site-packages/torch/nn/modules/conv.py", line 549, in _conv_forward
    return F.conv2d(
           ^^^^^^^^^
GET was unable to find an engine to execute this computation

**Comment by hjh0119** [comment]
https://github.com/ultralytics/ultralytics/issues/4060
